ext/odbc/php_odbc.c

```diff
@@ -2051,18 +2089,10 @@ PHP_FUNCTION(odbc_free_result)
 	result = Z_ODBC_RESULT_P(pv_res);
 	CHECK_ODBC_RESULT(result);
 
-	if (result->values) {
-		for (i = 0; i < result->numcols; i++) {
-			if (result->values[i].value) {
-				efree(result->values[i].value);
-			}
-		}
-		efree(result->values);
-		result->values = NULL;
+	if (result->conn_ptr) {
```

Comment:
This if is not even needed since `CHECK_ODBC_RESULT` ensures about the result being open